document/CoreFormat.mdk

```diff
@@ -775,26 +868,31 @@ A `result` object describes a single result produced by an analysis tool.
 
 ### `ruleId` property { #result-ruleId }
 
-A `result` object should contain a property named `ruleId` whose value is a string containing
+Depending on the circumstances, a `rule` object either shall, may, or shall not contain
+a property named `ruleId` whose value is a string containing
 the stable, opaque identifier for the rule that was evaluated to produce the result.
 
 ~ IsoExample
 `"CA2101"`
 ~
 
-Analysis tools that produce the SARIF format shall provide a `ruleId` for each `result`
-(with the possible exception of `result`s whose `kind` property is `note`; see (&sect;[#result-kind]).
+If the log was created by an analysis tool (as opposed to a conversion tool), and if the `level` property (&sect;[#result-level]) is anything other than `note`,
+then `ruleId` shall be present.
+
+If the `level` property of the `result` object is `note`,
+then the `ruleId` property may be present.
 
 Not all existing analysis tools emit the equivalent of a `ruleId` in their output.
 A conversion tool which converts the output of such an analysis tool to the SARIF format
-shall omit `ruleId` from the `result`s.
+shall not set the `ruleId` property, and in particular, it shall not attempt to
```

Comment:
Yes, I think it is clear, because the paragraph sets the context as "tools that do not emit the equivalent of a rule id".